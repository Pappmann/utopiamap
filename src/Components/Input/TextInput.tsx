import { useEffect, useState } from 'react'
import * as React from 'react'

type InputTextProps = {
  labelTitle?: string
  labelStyle?: string
  type?: string
  dataField?: string
  containerStyle?: string
  inputStyle?: string
  defaultValue?: string
  placeholder?: string
  autocomplete?: string
  updateFormValue?: (value: string) => void
}

export function TextInput({
  labelTitle,
  labelStyle,
  type,
  dataField,
  containerStyle,
  inputStyle,
  defaultValue,
  placeholder,
  autocomplete,
  updateFormValue,
}: InputTextProps) {
  const [inputValue, setInputValue] = useState<string>(defaultValue || '')

  useEffect(() => {
    setInputValue(defaultValue || '')
  }, [defaultValue])

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value
    setInputValue(newValue)
    if (updateFormValue) {
      updateFormValue(newValue)
    }
  }

  return (
    <div className={`tw-form-control ${containerStyle}`}>
      {labelTitle ? (
        <label className='tw-label'>
          <span className={`tw-label-text tw-text-base-content ${labelStyle}`}>{labelTitle}</span>
        </label>
      ) : null}
      <input
        required
        type={type || 'text'}
        name={dataField}
        value={inputValue}
        placeholder={placeholder || ''}
        autoComplete={autocomplete}
        onChange={handleChange}
        className={`tw-input tw-input-bordered tw-w-full ${inputStyle || ''}`}
      />
    </div>
  )
}
