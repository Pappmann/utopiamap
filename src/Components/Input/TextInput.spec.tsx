import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import { TextInput } from './TextInput'

describe('<TextInput />', () => {
  it('renders properly', () => {
    const wrapper = render(<TextInput />)
    expect(wrapper.container.firstChild).toMatchSnapshot()
  })
})
