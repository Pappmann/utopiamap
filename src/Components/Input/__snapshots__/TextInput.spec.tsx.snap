// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TextInput /> > renders properly 1`] = `
<div
  class="tw-form-control undefined"
>
  <input
    class="tw-input tw-input-bordered tw-w-full "
    placeholder=""
    required=""
    type="text"
    value=""
  />
</div>
`;
